### PR TITLE
Enable swagger-ui for all services

### PR DESCRIPTION
**Pull request by sudo0rw3ll**
Hello team,
During previous days I was battling some anomalies which would occur when deploying
services with swagger configuration included. I will describe the problem and solution in the section down below:

1. Checked for springdoc dependency, both services already had it.
2. Wrote SwaggerConfiguration for both services
3. Ran services and started testing everything on localhost. That's where I occured first error.
I wasn't able to access /swagger-ui.html and other paths which openapi required so I fixed this
by changing SpringSecurityConfig class to permit access to needed resources.
4. Github actions deployed changes to PROD
5. Second error occured when aiming /swagger-ui on PROD_IP:PORT because swagger page contained only partial routes.
6. This was solved by adding checkout actions to the deployment file, and making github actions to switch to current branch and pull active changes of the repositories.
7. Also, I have added one line to the config which is used to tell springdoc where to look for controller classes.

Current deployment enables devs to navigate only to user-service's_ip:8080/swagger-ui.html since SpringSecurityConfig was not implemented for market service. This discrepancy leaves us with no possibility of interacting with market-service's swagger API.

Long story short, swagger config is also added for market service. Once its SecurityConfiguration is implemented, swagger-ui will be accessible for market-service also.

If there are any questions, feel free to contact me.

Best,
Vid

### CODE REVIEW

File: user-service/src/main/java/rs/edu/raf/banka1/configuration/SpringSecurityConfig.java [hunk 1]
```diff
@@ -41,7 +41,11 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .authorizeHttpRequests((authz) ->
                                 authz
 //                            .requestMatchers(HttpMethod.OPTIONS, "/**").permitAll()
-                                        .requestMatchers(antMatcher("/auth/**")).permitAll()
+                                        .requestMatchers(antMatcher("/auth/**"),
```
Comment (ijevtic) [inline user-service/src/main/java/rs/edu/raf/banka1/configuration/SpringSecurityConfig.java:44]: ajde dodaj i za onu activate token putanju

**Review by ijevtic — CHANGES_REQUESTED** [review]

File: user-service/src/main/java/rs/edu/raf/banka1/configuration/SpringSecurityConfig.java (hunk 1 above)
Comment (ijevtic) [reply user-service/src/main/java/rs/edu/raf/banka1/configuration/SpringSecurityConfig.java:44]: /user/activate/**

Comment (sudo0rw3ll) [reply user-service/src/main/java/rs/edu/raf/banka1/configuration/SpringSecurityConfig.java:44]: image

Can you just check with current version of the source if these permissions are present because when I did:
`git pull upstream sprint-2` I got these from the picture above.

Comment (ijevtic) [reply user-service/src/main/java/rs/edu/raf/banka1/configuration/SpringSecurityConfig.java:44]: no, its not present, it is a bug, when user sets password it doesn't have jwt token
